packages/forms-web-app/src/dynamic-forms/validator/string-validator.js

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable no-unused-vars */
+const { body } = require('express-validator');
+const BaseValidator = require('./base-validator.js');
+
+class StringValidator extends BaseValidator {
+	regex = {
+		regex: '',
+		regexMessage: ''
+	};
+	minLength = {
+		minLength: 0,
+		minLengthMessage: ''
+	};
+	maxLength = {
+		maxLength: 0,
+		maxLengthMessage: ''
+	};
+	constructor({ regex, minLength, maxLength } = {}) {
+		super();
+
+		if (!regex && !minLength && !maxLength) throw new Error('Nothing to do!');
+		if (regex && regex.regex) this.regex = regex;
+		if (minLength && minLength.minLength) this.minLength = minLength;
+		if (maxLength && maxLength.maxLength) this.maxLength = maxLength;
+	}
+	validate(questionObj) {
+		const regexValidator = (chain) =>
+			chain.matches(new RegExp(this.regex.regex)).withMessage(this.regex.regexMessage);
+		const minLengthValidator = (chain) =>
```

Comment:
This would maybe be readable as a traditional express validator chain fluent declaration like
```
chain.matches(new RegExp(this.regex.regex)).withMessage(this.regex.regexMessage)
.isLength({ min: this.minLength.minLength })
.withMessage(this.minLength.minLengthMessage)
.isLength({ max: this.maxLength.maxLength })
.withMessage(this.maxLength.maxLengthMessage);

```
Theres a good example here https://github.com/jayeshchoudhary/express-validator-example/blob/main/validations/user.validation.js

Reply:
I think you missed the point of what I was doing here. The chain is built dynamically based on inputs.

Reply:
We want it to always be a complete chain with either default values or by leveraging the optional syntax of the express validator as suggested in the ticket, the code example above would achieve the same effect but with less code and more consistency with the express validator docs and existing string evaluation in the code base.

Reply:
Please read the unit tests

Reply:
Most of the operators for express validator seem to be geared around branching within a chain based on request data. Since this is configuration based, it's a slightly different use-case.

Either this or some custom chain checks should work fine.